src/NUnitEngine/nunit.engine/Services/TestSelectionParser.cs

```diff
@@ -0,0 +1,275 @@
+﻿// ***********************************************************************
+// Copyright (c) 2015 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using NUnit.Common;
+
+namespace NUnit.Engine
+{
+    public class TestSelectionParser
+    {
+        private Tokenizer _tokenizer;
+
+        private readonly Token LPAREN = new Token(TokenKind.Symbol, "(");
+        private readonly Token RPAREN = new Token(TokenKind.Symbol, ")");
+        private readonly Token COMMA = new Token(TokenKind.Symbol, ",");
+        private readonly Token AND_OP1 = new Token(TokenKind.Symbol, "&");
+        private readonly Token AND_OP2 = new Token(TokenKind.Symbol, "&&");
+        private readonly Token OR_OP1 = new Token(TokenKind.Symbol, "|");
+        private readonly Token OR_OP2 = new Token(TokenKind.Symbol, "||");
```

Comment:
Possibly. I've noticed that a lot of scripting languages support = & and | but I figured our users would want to use the C# versions. Of course, VB people will like the single char versions. :-)